app/models/mixins/authentication_mixin.rb

```diff
@@ -451,6 +463,25 @@ def assign_nested_authentication(attributes)
 
   private
 
+  def set_authentication_for_has_one_type
+    auth_attrs = attributes[:authentication_attributes]
+    if auth_attrs && auth_attrs[:userid] && auth_attrs[:password]
+      attributes[:type]          = self.class.has_one_type
+      attributes[:resource_type] = resource_type_for_has_one
+    end
+  end
+
+  def resource_type_for_has_one
```

Comment:
Possibly could name this something more generic, but this is just following the same logic as is found in `update_authentication` a bit up in this file when determining how to build the credential (`cred`) object.